### PR TITLE
Add release notes for 7.12.1

### PR DESCRIPTION
**Pull request by AthenaEryma**
This commit adds release notes for 7.12.1.

There were no flagged release highlights for Elasticsearch for this release.

[Preview link](https://elasticsearch_71703.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.12/release-notes-7.12.1.html) (once the build finishes)

### CODE REVIEW

**Review by lcawl — APPROVED** [review]
ML content LGTM, thanks!

**Comment by gwbrown** [comment]
@lcawl Thanks for the review and for adding the ML notes. A couple questions (this is the first time I'm doing the release manager thing):
1) Should I have done something differently to include the ML release notes? Or is it expected that those will needed to be added separately?
2) Is this good to merge now, or do we need another review? I ask only because you specified that the ML content looks good.

File: docs/reference/release-notes/7.12.asciidoc
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:63]:
```suggestion
* Enable setting master node timeout in Watcher start/stop Requests {es-pull}70425[#70425]
```

File: docs/reference/release-notes/7.12.asciidoc
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:86]: I think the script was erroneously pulling these extra issues from the code example:
```suggestion
* Resolve attributes recursively for improved subquery support {es-pull}69765[#69765] (issue:  {es-issue}67237[#67237])
```

File: docs/reference/release-notes/7.12.asciidoc
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:97]:
```suggestion
* Fix source only snapshot permanently broken on broken `_snapshot` directory {es-pull}71459[#71459]
```

File: docs/reference/release-notes/7.12.asciidoc
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:103]:
```suggestion
* Reduce memory use of parallel Azure blob deletes {es-pull}71330[#71330] (issue: {es-issue}71267[#71267])
```

File: docs/reference/release-notes/7.12.asciidoc
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:104]:
```suggestion
* Stop blocking snapshot deletes due to concurrency limits {es-pull}71050[#71050]
```

**Comment by lcawl** [comment]
> 1. Should I have done something differently to include the ML release notes? Or is it expected that those will needed to be added separately?

Unfortunately, this is always a manual process. The ML writers do it, so it's not something the release coordinator has to worry about. 

> 2. Is this good to merge now, or do we need another review? I ask only because you specified that the ML content looks good.

Typically an Elasticsearch writer reviews as well, but I think most of them are away so I did a quick check of the rest of the PRs too and submitted some more comments.

**Comment by gwbrown** [comment]
@lcawl Thanks both the answers and the review are much appreciated!

**Pull request merged**
